client/src/components/createBlog/create.js

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import FormErrors from '../errors/formErrors';
+import Constants from '../../helpers/constants';
+import FormElement from '../formElements/formElement';
+import './create.css';
+import { connect } from 'react-redux';
+import * as blogActions from '../../store/actions/blog'
+
+
+class Create extends Component {
+
+  constructor() {
+    super();
+    this.state = {
+      title: '',
+      author: '',
+      body: '',
+      formErrors: { title: '', author: '', body: '' },
+      titleValid: false,
+      bodyValid: false,
+      formValid: false
+    };
+  }
+
+  onStateUpdated = (name, value) => { this.validateField(name, value) };
+
+  onChange = (e) => {
+    const { name, value } = e.target;
+    this.setState({ [name]: value },
+      this.onStateUpdated(name, value));
+  }
+
+  onSubmit = (e) => {
+    e.preventDefault();
+
+    const { title, body } = this.state;
+    const author = this.props.username;
+
+    this.props.createBlog(JSON.stringify({ title, author, body }), () => {
+      this.props.history.push("/blogs");
+    });
+  }
+
+  validateField(fieldName, value) {
+    let { formErrors, titleValid, bodyValid } = this.state;
+
+    switch (fieldName) {
+      case 'title':
+        titleValid = value.length >= Constants.MIN_LENGTH && value.length <= Constants.MAX_LENGTH;
+        formErrors.title = titleValid ? '' : ' is invalid';
+        break;
+      case 'body':
+        bodyValid = value.length > Constants.ZERO_LENGTH;
+        formErrors.body = bodyValid ? '' : ' is invalid';
+        break;
+      default:
+        break;
+    }
+    this.setState({
+      formErrors: formErrors,
+      titleValid: titleValid,
+      bodyValid: bodyValid
+    }, this.validateForm);
+  }
+
+  validateForm() {
+    this.setState({ formValid: this.state.titleValid && this.state.bodyValid });
+  }
+
+  errorClass(error) {
```

Comment:
This uses more that 2 time.
Export into util method and import after that, please.
And write test for it